src/dg/artificial_dissipation.h

```diff
@@ -0,0 +1,117 @@
+#ifndef _ARTIFICIALDISSIPATION_
+#define _ARTIFICIALDISSIPATION_
+
+#include "physics/physics.h"
+#include "physics/convection_diffusion.h"
+#include "physics/navier_stokes.h"
+#include "parameters/all_parameters.h"
+#include "parameters/parameters_manufactured_solution.h"
+#include "ADTypes.hpp"
+
+ namespace PHiLiP{
+
+ template <int dim, int nstate>
```

Comment:
How come this class is not templated with `typename real`?

Reply:
The functions inside Artificial Dissipation Class are overloaded for all typenames (to avoid creating multiple artificial dissipation pointers and changing function parameters in DGWeak).